### PR TITLE
Changes in documentation. Rephrasing, fixed examples, standarized notation, etc.

### PR DESCRIPTION
**Pull request by grudloff**

### CODE REVIEW

File: doc/supervised.rst [hunk 1]
```diff
@@ -235,7 +235,7 @@ the sum of probability of being correctly classified:
 
 Local Fisher Discriminant Analysis (:py:class:`LFDA <metric_learn.LFDA>`)
 
-`LFDA` is a linear supervised dimensionality reduction method. It is
+`LFDA` is a linear supervised dimensionality reduction method witch effectively combines the ideas of `Linear Discriminant Analysis <https://en.wikipedia.org/wiki/Linear_discriminant_analysis>` and Locality-Preserving Projection . It is
```
Comment (perimosocordiae) [inline doc/supervised.rst]: witch -> which

File: doc/getting_started.rst [hunk 2]
```diff
@@ -20,7 +20,8 @@ Alternately, download the source repository and run:
 **Optional dependencies**
 
 - For SDML, using skggm will allow the algorithm to solve problematic cases
-  (install from commit `a0ed406 <https://github.com/skggm/skggm/commit/a0ed406586c4364ea3297a658f415e13b5cbdaf8>`_).
+  (install from commit `a0ed406 <https://github.com/skggm/skggm/commit/a0ed406586c4364ea3297a658f415e13b5cbdaf8>`_). 
+- ``pip install ''git+https://github.com/skggm/skggm.git@a0ed406586c4364ea3297a658f415e13b5cbdaf8''`` to install the required version of skggm from PyPl.
```
Comment (perimosocordiae) [inline doc/getting_started.rst]: Single quotes here, rather than `''`.

File: doc/index.rst [hunk 3]
```diff
@@ -52,7 +52,7 @@ Documentation outline
 
    auto_examples/index
 
-:ref:`genindex` | :ref:`modindex` | :ref:`search`
+:ref:`genindex` | :doc:`Modules <./metric_learn>` | :ref:`search`
```
Comment (perimosocordiae) [inline doc/index.rst]: What does this do?

File: doc/getting_started.rst [hunk 4]
```diff
@@ -10,7 +10,7 @@ Run ``pip install metric-learn`` to download and install from PyPI.
 Alternately, download the source repository and run:
 
 -  ``python setup.py install`` for default installation.
--  ``python setup.py test`` to run all tests.
+-  ``pytest test`` to run all tests.
```
Comment (bellet) [inline doc/getting_started.rst]: Indeed it seems that `python setup.py test` has been deprecated:
https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite
They now recommend to use `tox` instead, which seems quite heavy

So I am happy with the proposed solution of simply suggesting to run `pytest`

File: doc/getting_started.rst (hunk 2 above)
Comment (bellet) [reply doc/getting_started.rst]: This could also be added to the README

File: doc/supervised.rst [hunk 5]
```diff
@@ -261,18 +261,18 @@ where
     \,\,\mathbf{A}_{i,j}(1/n-1/n_l) \qquad y_i = y_j\end{aligned}\right.\\
 
 here :math:`\mathbf{A}_{i,j}` is the :math:`(i,j)`-th entry of the affinity
-matrix :math:`\mathbf{A}`:, which can be calculated with local scaling methods.
+matrix :math:`\mathbf{A}`:, which can be calculated with local scaling methods, `n` and `n_l` are the total number of points and the number of points per cluster `l`.
```
Comment (bellet) [inline doc/supervised.rst]: add "respectively"

File: metric_learn/lsml.py [hunk 6]
```diff
@@ -290,6 +292,16 @@ class LSML_Supervised(_BaseLSML, TransformerMixin):
...
 
+  Examples
+  --------
+  >>> from metric_learn import ITML_Supervised
+  >>> from sklearn.datasets import load_iris
+  >>> iris_data = load_iris()
+  >>> X = iris_data['data']
+  >>> Y = iris_data['target']
+  >>> itml = ITML_Supervised(num_constraints=200)
+  >>> itml.fit(X, Y)
```
Comment (bellet) [inline metric_learn/lsml.py]: this should be LSML_Supervised, not ITML_Supervised

File: metric_learn/rca.py [hunk 7]
```diff
@@ -62,13 +62,16 @@ class RCA(MahalanobisMixin, TransformerMixin):
...
+  >>> from metric_learn import RCA
+  >>> pairs = [[[1.2, 7.5], [1.3, 1.5]],
+  >>>         [[6.4, 2.6], [6.2, 9.7]],
+  >>>         [[1.3, 4.5], [3.2, 4.6]],
+  >>>         [[6.2, 5.5], [5.4, 5.4]]]
+  >>> y = [1, 1, -1, -1]
+  >>> # in this task we want points where the first feature is close to be
+  >>> # closer to each other, no matter how close the second feature is
+  >>> rca = RCA()
+  >>> rca.fit(pairs, y)
```
Comment (bellet) [inline metric_learn/rca.py]: This example is wrong: RCA learns from chunks, not from pairs

Comment (grudloff) [reply metric_learn/rca.py]: I took this example from the doc/weakly_supervised.rst RCA section. I will fix them both then.

Comment (grudloff) [reply metric_learn/rca.py]: Then it would also be necessary to make a new section in Weakly Supervised Metric Learning called "Learning on chunks"? Right now RCA is under "Learning on pairs".

File: doc/index.rst (hunk 3 above)
Comment (bellet) [reply doc/index.rst]: This page is already linked a bit above on the index page, so perhaps we could simply remove it?
Or, fix the problem preventing the generation of this file

File: metric_learn/rca.py (hunk 7 above)
Comment (bellet) [reply metric_learn/rca.py]: Yes, there is an issue on that, see #237 and comments therein
Since the idea of chunks is very specific to RCA, with a small abuse of language we were considering listing it under the supervised algorithms, even though the supervision is weaker

File: README.rst [hunk 8]
```diff
@@ -26,6 +26,7 @@ metric-learn contains efficient Python implementations of several popular superv
 
 - For SDML, using skggm will allow the algorithm to solve problematic cases
   (install from commit `a0ed406 <https://github.com/skggm/skggm/commit/a0ed406586c4364ea3297a658f415e13b5cbdaf8>`_).
+- ``pip install 'git+https://github.com/skggm/skggm.git@a0ed406586c4364ea3297a658f415e13b5cbdaf8'`` to install the required version of skggm from PyPl.
```
Comment (bellet) [inline README.rst]: - Please do not create a new list item as this is still about the skggm optional dependency
- This does not install from PyPI but from GitHub

File: doc/getting_started.rst [hunk 9]
```diff
@@ -20,7 +20,8 @@ Alternately, download the source repository and run:
 **Optional dependencies**
 
 - For SDML, using skggm will allow the algorithm to solve problematic cases
-  (install from commit `a0ed406 <https://github.com/skggm/skggm/commit/a0ed406586c4364ea3297a658f415e13b5cbdaf8>`_).
+  (install from commit `a0ed406 <https://github.com/skggm/skggm/commit/a0ed406586c4364ea3297a658f415e13b5cbdaf8>`_). 
+- ``pip install 'git+https://github.com/skggm/skggm.git@a0ed406586c4364ea3297a658f415e13b5cbdaf8'`` to install the required version of skggm from PyPl.
```
Comment (bellet) [inline doc/getting_started.rst]: same here

File: metric_learn/rca.py (hunk 7 above)
Comment (grudloff) [reply metric_learn/rca.py]: I think that it should remain in the weak learners section for consistency's sake, as there is a *_supervised version, even though the supervision is not so weak. Would it be worth it to make a section just for RCA learning on chunks?

File: doc/index.rst (hunk 3 above)
Comment (grudloff) [reply doc/index.rst]: I am not knowledgeable in sphinx so I can't find a quick fix, in the meantime, I will remove it.

File: metric_learn/rca.py (hunk 7 above)
Comment (bellet) [reply metric_learn/rca.py]: Sure, why not, although it is unlikely to apply to another algorithm other than RCA in the future
If you are willing to do this, please go ahead. Probably a separate PR would be best for clarity

Comment (grudloff) [reply metric_learn/rca.py]: Alright then, in the meantime I fixed the RCA examples as they are.

**Comment by bellet** [comment]
LGTM! Thanks for the changes and congrats for your first contribution @grudloff :-)
